src/query/execute_statement.c

```diff
@@ -17724,3 +17758,361 @@ do_insert_checks (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE ** class
+
+  obj_tmpl = dbt_create_object_internal ((MOP) server_class);
+  if (obj_tmpl == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto end;
+    }
+
+  /* server_name */
+  db_make_string (&value, server_name);
+  error = dbt_put_internal (obj_tmpl, "link_name", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* host */
+  db_make_string (&value, host);
+  error = dbt_put_internal (obj_tmpl, "host", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* port */
+  error = dbt_put_internal (obj_tmpl, "port", port_no);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* dbname */
+  db_make_string (&value, dbname);
+  error = dbt_put_internal (obj_tmpl, "db_name", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* username */
+  db_make_string (&value, username);
+  error = dbt_put_internal (obj_tmpl, "user_name", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* password */
+  db_make_string (&value, password);
+  error = dbt_put_internal (obj_tmpl, "password", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* properties */
+  db_make_string (&value, properties);
+  error = dbt_put_internal (obj_tmpl, "properties", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* owner */
+  db_make_object (&value, Au_user);
+  error = dbt_put_internal (obj_tmpl, "owner", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* comment */
+  db_make_string (&value, comment);
+  error = dbt_put_internal (obj_tmpl, "comment", &value);
+  pr_clear_value (&value);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  ret_obj = dbt_finish_object (obj_tmpl);
+  if (ret_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+    }
+  else if (server_object != NULL)
+    {
+      *server_object = ret_obj;
+    }
+
+end:
+  if (obj_tmpl != NULL && ret_obj == NULL)
+    {
+      dbt_abort_object (obj_tmpl);
+    }
+  AU_ENABLE (au_save);
+  return error;
+}
+
+int
+do_create_server (PARSER_CONTEXT * parser, PT_NODE * statement)
+{
+  DB_OBJECT *server_class = NULL, *server_object = NULL;
+  DB_IDENTIFIER server_obj_id;
+  DB_VALUE *pval = NULL;
+  char *server_name = NULL;
+  DB_VALUE port_no;
+  DB_DATA_STATUS data_stat;
+  const char *host = NULL;
+  const char *dbname = NULL;
+  const char *username = NULL;
+  const char *password = NULL;
+  const char *properties = NULL;
+  const char *comment = NULL;
+  int error = NO_ERROR;
+  int save;
+  bool au_disable_flag = false;
+
+  PT_CREATE_SERVER_INFO *si = &statement->info.create_server;
+
+  CHECK_MODIFICATION_ERROR ();
+
+  db_make_null (&port_no);
+  db_make_int (&port_no, 0);
```

Comment:
Why don't you assign the type of **port_no** to "unsigned short" or "unsigned integer" ?

Reply:
The type of the port column of _db_server is INTEGER.
Actually, it will not have a value beyond the range of the port number because valid value check is carried out.

Reply:
In accordance with your seperated  comments, we have simplified the processing parameters in the function.